src/main/java/io/axoniq/axonserver/connector/impl/FlowControlledBuffer.java

```diff
@@ -45,6 +46,9 @@
      */
     public FlowControlledBuffer(String clientId, int bufferSize, int refillBatch) {
         super(clientId, bufferSize, refillBatch);
+        if (terminalMessage() == null) {
```

Comment:
Calling a non-final method from a constructor is dangerous, as it may be overridden by subclasses and could be called before the constructor has finished.

Reply:
I moved the validation elsewhere, good catch